evaluator.activitydefinition/src/main/java/org/opencds/cqf/cql/evaluator/activitydefinition/r5/ActivityDefinitionProcessor.java

```diff
@@ -35,10 +41,42 @@ public ActivityDefinitionProcessor(Repository repository) {
     super(repository);
   }
 
-  // For library use
   @Override
-  public Resource resolveActivityDefinition(ActivityDefinition activityDefinition, String patientId,
-      String practitionerId, String organizationId) throws FHIRException {
+  @SuppressWarnings("unchecked")
+  public <R extends IBaseResource> R searchRepositoryByUrl(Class<R> theResourceType,
+      String theUrl) {
+    var searchResult = repository.search(Bundle.class, theResourceType, Searches.byUrl(theUrl));
+    if (!searchResult.hasEntry()) {
+      throw new FHIRException(String.format("No resource of type %s found for url: %s",
+          theResourceType.getSimpleName(), theUrl));
+    }
+
+    return (R) searchResult.getEntryFirstRep().getResource();
+  }
+
+  @Override
+  public ActivityDefinition resolveActivityDefinition(IIdType theId, String theCanonical,
```

Comment:
Question: I have seen this method resolveActivityDefinition in 3 different places. Are all the implementations the same, if so can they be abstracted to any super implementation. For me them looked exactly same unless I am not seeing something.

Even the method above searchRepositoryByUrl. 

Please correct me or let me know if we need to discuss.

Reply:
The resource types that are needed here are version specific.  Hence the reason for the abstract method only in the Base class and the very similar implementation in each version.

Reply:
The searchRepositoryByUrl function could potentially go into a version specific helper class and I briefly thought about doing that but decided to wait until we have a better place to put such things, i.e. once we have everything shifted from the evaluator packages into the cqf-fhir packages.